### PR TITLE
[docs][Charts] Give a general revision to the docs

### PR DESCRIPTION
**Pull request by danilo-leal**
- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/mui/mui-x/blob/HEAD/CONTRIBUTING.md#sending-a-pull-request).

This PR gives a general revision to the MUI X Charts space, proposing changes such as:
- Tweaking the page hierarchy structure by adding section dividers ("Available components", "Future components", and "Customization"). I believe this makes it easier to consume our different types of content.
- Copywriting and formatting changes to the Overview page.
- Fixing all demo instances that are meant to be a "Playground", so the design is consistent with all other MUI docs!
- Standardizing how we display planned future components to what we do on Core — removing the emojis from the title, adding the "Planned" chip, and tweaking the callout copywriting.

👉 https://deploy-preview-11249--material-ui-x.netlify.app/x/react-charts/

### CODE REVIEW

File: docs/.link-check-errors.txt [hunk 1]
```diff
@@ -17,3 +17,4 @@ Broken links found by `yarn docs:link-check` that exist:
 - https://mui.com/x/api/data-grid/data-grid/#slots
 - https://mui.com/x/api/date-pickers/date-picker/#slots
 - https://mui.com/x/introduction/licensing/#license-key-installation
+- https://mui.com/x/react-charts/#getting-started
```
[the remark this answers is not in the document]
Comment (alexfauquette) [reply docs/.link-check-errors.txt:20]: Yes, it's from the blog for v6. Here is the fix

https://github.com/mui/material-ui/pull/40043#issuecomment-1833347655

**Comment by alexfauquette** [comment]
A similar effort has already been merged on master and is cherry-picked on next #11239

File: docs/data/charts/funnel/funnel.md [hunk 2]
```diff
@@ -2,14 +2,13 @@
...
 ---
 
-# Charts - Funnel 🚧[<span class="plan-pro"></span>](/x/introduction/licensing/#pro-plan 'Pro plan')
+# Charts - Funnel [<span class="plan-pro"></span>](/x/introduction/licensing/#pro-plan 'Pro plan')
 
 <p class="description">Funnel charts allow to express quantity evolution along a process, such as audience engagement.</p>
 
 :::warning
-This feature isn't implemented yet. It's coming.
+The Funnel variant of the Charts component isn't available yet, but you can upvote [**this GitHub issue**](https://github.com/mui/mui-x/issues/7929) to see it arrive sooner.
```
[the remark this answers is not in the document]
Comment (danilo-leal) [reply docs/data/charts/funnel/funnel.md:10]: Yeah, was also reflecting on this — it said "feature" previously, which felt even smaller than "variant". 😅
Guess we can use @samuelsycamore's take here 🤙

**Comment by danilo-leal** [comment]
@alexfauquette, should I wait for you to finish/merge that other PR? Any recommendations? 😬

**Comment by alexfauquette** [comment]
The other PR is ready to be merged. It's a cherry-pick of another PR that has already been accepted

**Comment by danilo-leal** [comment]
Ok, I'll wait for that and then resolve any conflicts with this one!

**Comment by danilo-leal** [comment]
@flaviendelangle + @alexfauquette can y'all take a second look at this when possible? 😃

File: docs/data/charts/getting-started/getting-started.md [hunk 3]
```diff
@@ -32,44 +28,49 @@ pnpm add @mui/x-charts@next
 </codeblock>
 
 :::info
-With Next.js you might face the following error:
+The `next` tag is used to download the latest, **pre-release**, v7 version.
+:::
```
Comment (alexfauquette) [inline docs/data/charts/getting-started/getting-started.md:32]: The place was to be coherent with other packages. But this one is fine too

**Review by alexfauquette — APPROVED** [review]
👌

File: docs/data/charts/getting-started/getting-started.md (hunk 3 above)
Comment (danilo-leal) [reply docs/data/charts/getting-started/getting-started.md]: I'll make sure the others are consistent as well! 
Took advantage of this callout, though, to clarify even further about the `next` tag 😬

File: docs/src/modules/components/ChartComponentsGrid.js [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+import Typography from '@mui/material/Typography';
+import Link from 'docs/src/modules/components/Link';
+
+function components() {
+  return [
+    {
+      title: 'Bar Chart',
+      srcLight: '/static/x/component-illustrations/bar-light.png',
+      srcDark: '/static/x/component-illustrations/bar-dark.png',
+      href: '/x/react-charts/bar/',
```
Comment (oliviertassinari) [inline docs/src/modules/components/ChartComponentsGrid.js:15]: Oops https://next.mui.com/x/react-charts/bar/

Fixed in 1a131e0b48e674aae755d21e8f40c554d1f63a07.

Comment (danilo-leal) [reply docs/src/modules/components/ChartComponentsGrid.js:15]: Thank you!

**Pull request merged**